community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/ZonedDateTimeSchemaKey.java

```diff
@@ -82,7 +82,9 @@ public int compareValueTo( ZonedDateTimeSchemaKey other )
         if ( compare == 0 )
         {
             compare = Integer.compare( nanoOfSecond, other.nanoOfSecond );
-            if ( compare == 0 && hasValidTimeZone() && other.hasValidTimeZone() )
+            if ( compare == 0 &&
+                    TimeZones.validZoneOffset( zoneOffsetMinutes * 60 ) &&
+                    TimeZones.validZoneOffset( other.zoneOffsetMinutes * 60 ) )
```

Comment:
Wouldn't it be easier to check for something like `!sameTimeZone( this, other )`. I don't understand what this validity check has to do with anything.

Unless the comment in the Javadoc of this class is correct and we order named zones separately from offset zones. Which is not the order the commit comment says.

Reply:
The comment is wrong. I have not realized that for once we actually do have a comment on a class and forgot to change this. 

W.r.t to the check: It will succeed in all cases (named zones and offsets) where the PageCursor does not return garbage (as I understand). Then, the Value Comparator takes care of ordering them as implemented in `DateTimeValue.unsafeCompareTo`

Reply:
Then I wonder if this is the correct behaviour in the case of garbage being returned. Isn't the contract to retry in the case of invalid reads?